tests/integration/distribution.go

```diff
@@ -690,197 +696,92 @@ func (s *CCVTestSuite) TestIBCTransferMiddleware() {
-			},
-			sdk.DecCoins{
-				sdk.DecCoin{
-					Denom:  sdk.DefaultBondDenom,
-					Amount: math.LegacyNewDecWithPrec(995, 1),
-				},
-			},
-			false,
-		},
-	}
-
+func (s *CCVTestSuite) TestAllocateTokens() {
+	// set up channel and delegate some tokens in order for validator set update to be sent to the consumer chain
+	s.SetupAllCCVChannels()
 	providerKeeper := s.providerApp.GetProviderKeeper()
 	bankKeeper := s.providerApp.GetTestBankKeeper()
 	distributionKeeper := s.providerApp.GetTestDistributionKeeper()
+	accountKeeper := s.providerApp.GetTestAccountKeeper()
 
-	chainID := s.consumerChain.ChainID
+	getDistrAcctBalFn := func(ctx sdk.Context) sdk.DecCoins {
+		bal := bankKeeper.GetAllBalances(ctx, accountKeeper.GetModuleAccount(ctx, distrtypes.ModuleName).GetAddress())
+		return sdk.NewDecCoinsFromCoins(bal...)
+	}
 
-	for _, tc := range testCases {
-		s.Run(tc.name, func() {
-			ctx, _ := s.providerCtx().CacheContext()
-			// fund consumer rewards pool
-			bankKeeper.SendCoinsFromAccountToModule(
-				ctx,
-				s.providerChain.SenderAccount.GetAddress(),
-				providertypes.ConsumerRewardsPool,
-				tc.rewardsPool,
-			)
+	totalRewards := sdk.Coins{sdk.NewCoin(sdk.DefaultBondDenom, math.NewInt(100))}
 
-			// update consumer rewars allocation
-			providerKeeper.SetConsumerRewardsAllocation(
-				ctx,
-				chainID,
-				providertypes.ConsumerRewardsAllocation{
-					Rewards: tc.rewardsAlloc,
-				},
-			)
+	// fund consumer rewards pool
+	bankKeeper.SendCoinsFromAccountToModule(
+		s.providerCtx(),
+		s.providerChain.SenderAccount.GetAddress(),
+		providertypes.ConsumerRewardsPool,
+		totalRewards,
+	)
 
-			// store pool balance
-			oldPool := bankKeeper.GetAllBalances(
-				ctx,
-				distributionKeeper.GetDistributionAccount(ctx).GetAddress(),
-			)
+	// Allocate rewards evenly between consumers
+	rewardsPerConsumer := totalRewards.QuoInt(math.NewInt(int64(len(s.consumerBundles))))
+	for chainID := range s.consumerBundles {
+		// update consumer allocation
+		providerKeeper.SetConsumerRewardsAllocation(
+			s.providerCtx(),
+			chainID,
+			providertypes.ConsumerRewardsAllocation{
+				Rewards: sdk.NewDecCoinsFromCoins(rewardsPerConsumer...),
+			},
+		)
+	}
 
-			coinsTransferred, _ := providerKeeper.GetConsumerRewardsAllocation(ctx, chainID).Rewards.TruncateDecimal()
-			// // transfer consumer rewards to distribution module
-			// coinsTransferred, err := providerKeeper.TransferConsumerRewardsToDistributionModule(
-			// 	ctx,
-			// 	chainID,
-			// )
-			// if tc.expErr {
-			// 	s.Require().Error(err)
-			// 	return
-			// }
-
-			// check remaining consumer rewards allocation
-			expCoinTransferred, expRemaining := tc.rewardsAlloc.TruncateDecimal()
-			s.Require().Equal(expCoinTransferred, coinsTransferred)
-
-			s.Require().Equal(
-				expRemaining,
-				providerKeeper.GetConsumerRewardsAllocation(ctx, chainID).Rewards,
-			)
+	// Iterate over the validators and
+	// store their current voting power and outstanding rewards
```

Comment:
```suggestion
	// store their outstanding rewards
```
Voting power is not stored anymore (?)